examples/tests/BUILD

```diff
@@ -0,0 +1,13 @@
+load("@rules_python//python:defs.bzl", "py_test")
+
+package(default_visibility = ["//visibility:public"])
+
+py_test(
+    name = "extension_test",
+    srcs = ["extension_test.py"],
+    data = [
+        "//extension:adder_extension",
```

Comment:
It seems you want is rather a py_library to be loaded at runtime (data) for the test, we had the same need.
You can simply create a py_library after the pybind_extension and put the extension with:
```
  py_library(
        name = name_of_extension,
        data = [name_of_extension + ".so"],
        ...
        imports = imports,
        visibility = visibility,
```
No need to build anything else. This py_library has no `srcs`, it is only for runtime.